src/hooks/useFetch.js

```diff
@@ -51,15 +52,27 @@ export const useFetch = (url, options = {}, timeout = CONNECTION_TIMEOUT_PERIOD)
             if (!responseData.length) throw new Error(ERROR_CODES.RESPONSE_NO_RECORDS);
             if (isMounted.current) setData(responseData);
           } else {
+            const connectionError = new Error(ERROR_CODES.CONNECTION_FAILURE);
+            const authorizationError = new Error(ERROR_CODES.INVALID_PASSWORD);
             switch (status) {
               case 400:
-              default:
-                throw new Error(ERROR_CODES.CONNECTION_FAILURE);
+                throw connectionError;
               case 401:
-                throw new Error(ERROR_CODES.INVALID_PASSWORD);
+                throw authorizationError;
+              default:
+                bugsnagClient.notify(connectionError, content => {
+                  content.requestUrl = url;
+                  content.requestHeaders = headers;
+                  content.responseStatus = status;
+                });
+                throw connectionError;
             }
           }
         } catch (responseError) {
+          bugsnagClient.notify(responseError, content => {
+            content.requestUrl = url;
+            content.requestHeaders = headers;
```

Comment:
I'm not a bugsnag expert, so could be wrong but this one here that was added: 

https://github.com/openmsupply/mobile/blob/d00f1274640307c64fadd8a18df877ba20375954/src/sync/Synchroniser.js#L94-L100

Uses the `metadata` field, and I wouldn't have done it like this unless I basically had to, I imagine! 

This gives an error like this one in the bugsnag (hope this link works): https://app.bugsnag.com/sustainable-solutions-nz-ltd/msupply-mobile/errors/5e4ca18b2bd15d0019031ab6?filters[event.since][0]=2020-02-15T02%3A38%3A24.109Z&filters[search][0][type]=eq&filters[search][0][value]=%22sync%20error%22&filters[error.status][0]=fixed

if it doesn't:

![image](https://user-images.githubusercontent.com/35858975/79293400-be37c700-7f27-11ea-9a57-66a12372914c.png)

Basically the `metaData` field is used to populate the `error` tab (which isn't there if you supply no `metaData` and the `context` is used for the .. 'name'?

The bugsnag docs suck imo! https://docs.bugsnag.com/platforms/react-native/react-native/#sending-diagnostic-data

Reply:
Hmm, I was lazy and just referenced other bugsnags in the codebase. Looks like you're right though. 

I think there might be a few broken calls in other places (the places I looked at :laughing:)? Maybe I'm missing something... if not we might need to make an issue for this?

`outgoingSyncUtils`:

    bugsnagClient.notify(error, content => {
      content.syncSite = UIDatabase.getSetting(SETTINGS_KEYS.SYNC_SITE_NAME);
      content.record = syncOutRecord;
    });
    ...
    bugsnagClient.notify(error, content => {
      content.syncSite = UIDatabase.getSetting(SETTINGS_KEYS.SYNC_SITE_NAME);
      content.record = syncOutRecord;
    });

`outgoingSyncUtils`:

    bugsnagNotify('Failed parsing response from server', response);
    ...
    bugsnagNotify(responseJson.error, response);


`synchroniser`:

    bugsnagNotify('Failed parsing response from server', response);
      throw new Error('Unexpected response from sync server');
    }
    ...
    bugsnagNotify(responseJson.error, response);
    throw new Error('Server rejected pushed records');

Reply:
I mean it looks like yeah, those are broken. Benefit of the doubt: it was probably an old bugsnag version and the calls never got migrated. To confirm, would need to either trigger, or find snags of those in the big list of snagged errors and check if any of those details are there.

Reply:
63fcda17.